src/Arrays/MaxContiguousSubArrayLength.java

```diff
@@ -0,0 +1,40 @@
+package Arrays;
+
+import java.util.PriorityQueue;
+
+public class MaxContiguousSubArrayLength {
+	
+	public static int findMaxLength(int[] nums) {
+		int count_0 = 0;
+		int count_1 = 0;
+		int max_length = 0;
+		for (int i=nums.length-1; i>=0; i--){
+			if (nums[i] == 0) {
+				count_0++;
+			}
+			else {
+				count_1++;
+			}
+			if (count_0 == count_1) {
+				if(max_length <= count_0) {
+					max_length = count_0*2;
+				}
+			}
+		}
+		if (max_length == 0 ) {
+            if (count_0 > count_1) {
```

Comment:
I believe your solution might fail if the input is "010100000101010"

Reply:
So is this a valid solution?

Reply:
If it fails for an input then why do we have it here?

Reply:
vignesh has to confirm. I do not know why he added it here.

Reply:
I did not add the solution for max contiguous array - i only added for valid parenthesis

Reply:
Lol wait i think that was my solution, how did it come in. It was a draft, Narayanan was the correct solution. Ok, I think it might be because of some fault with the rebase.